### PR TITLE
Add unit tests article for plugins

### PR DESCRIPTION
**Pull request by zbranzov**

### CODE REVIEW

File: plugins/unit-tests.md [hunk 1]
```diff
@@ -0,0 +1,92 @@
+---
+title: Unit Tests
+description: A guide describing how easy is to write unit tests for plugin created using nativescript-plugin-seed.
+position: 99
+slug: unit-tests
+environment: nativescript
```
Comment (radeva) [inline plugins/unit-tests.md]: If the article is relevant to both nativescript and angular, remove the environment

File: plugins/unit-tests.md [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+---
+
+# Unit Tests
+
+Writting unit tests for a plugin that is developed using the [nativescript-plugin-seed](https://github.com/NativeScript/nativescript-plugin-seed) is as simple as building the plugin itself. Before we continue, take a look at [Building Plugins]({% slug building-plugins %}) topic if you have missed that.
```
Comment (radeva) [inline plugins/unit-tests.md]: typo: Writting -> Writing

File: plugins/unit-tests.md [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+my-plugin
+├── demo
+|   └── app
+|       └── tests
+└── src
+     
+```
+
+The whole structure comes after you init your plugin from the [nativescript-plugin-seed](https://github.com/NativeScript/nativescript-plugin-seed) so you do not need to care about anything.
+
```
Comment (radeva) [inline plugins/unit-tests.md]: "The whole structure comes after you init" - I'm not sure how this sounds for native speakers. @tjvantoll can confirm if it is ok. My suggestion is 'The whole test structure comes for you out of the box when you init your..."

File: plugins/unit-tests.md [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+npm run test.ios
+```
+
+Be sure that you have available android/ios device or simulator.
+
+> **NOTE**: Using the npm scripts to run your tests is the best option, but in case of debugging purposes, where you want to output some console logs, you will have to navigate to `demo` folder and run `tns test android` or `tns test ios`.
+
+## Continuous Integration
+
+By starting from the [nativescript-plugin-seed](https://github.com/NativeScript/nativescript-plugin-seed) you get out of the box fully functional `.travis.yml` file ready to run your unit tests on Android and iOS in [Travis CI](https://travis-ci.org/). All you have to do is to log in to Travis CI, activate your repository and make sure any of `Build pushes` and `Build pull requests` is [enabled](https://docs.travis-ci.com/images/settings-env-vars.png).
```
Comment (radeva) [inline plugins/unit-tests.md]: At the end you can link the article "Ensure Plugins Quality": https://github.com/NativeScript/docs/blob/master/plugins/ensure-plugins-quality.md

**Review by radeva — CHANGES_REQUESTED** [review]

File: plugins/unit-tests.md [hunk 5]
```diff
@@ -0,0 +1,92 @@
+---
+title: Unit Tests
+description: A guide describing how easy is to write unit tests for plugin created using nativescript-plugin-seed.
```
Comment (tjvantoll) [inline plugins/unit-tests.md]: how easy it is to write unit tests for plugins created using the NativeScript plugin seed.

File: plugins/unit-tests.md [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+position: 99
+slug: unit-tests
```
Comment (tjvantoll) [inline plugins/unit-tests.md]: Probably want to go with plugin-unit-tests here.

File: plugins/unit-tests.md [hunk 7]
```diff
@@ -0,0 +1,92 @@
...
+Writting unit tests for a plugin that is developed using the [nativescript-plugin-seed](https://github.com/NativeScript/nativescript-plugin-seed) is as simple as building the plugin itself. Before we continue, take a look at [Building Plugins]({% slug building-plugins %}) topic if you have missed that.
+
+In this article you will find:
+- [Test Implementation](#test-implementation)
+- [Test Execution](#test-execution)
+- [Continuous Integration](#continuous-integration)
+
+## Test Implementation
+
+You have completed your plugin and it works great, but how you can be sure that every other change applied to the code base will not break some functionality and how to easily check plugin's state. Here, unit tests come to assistance with their best feature - 'fast execution'.
```
Comment (tjvantoll) [inline plugins/unit-tests.md]: check a plugin’s state

File: plugins/unit-tests.md [hunk 8]
```diff
@@ -0,0 +1,92 @@
...
+
+The starting point of writting unit tests is the `tests` folder in your demo app directory. There you will find `tests.js` file containing sample [Jasmine](https://jasmine.github.io/) tests.
```
Comment (tjvantoll) [inline plugins/unit-tests.md]: writting --> writing

File: plugins/unit-tests.md (hunk 3 above)
Comment (tjvantoll) [reply plugins/unit-tests.md]: The current wording is ok. I get what you mean and that’s the most important thing. Since you asked though, this is how I would personally word this:

The NativeScript plugin seed builds this structure automatically when you initialize your plugin, so don’t worry about creating these files and folders manually.

File: plugins/unit-tests.md [hunk 9]
```diff
@@ -0,0 +1,92 @@
...
+├── demo
+|   └── app
+|       └── tests
+└── src
+     
+```
+
+The whole structure comes after you init your plugin from the [nativescript-plugin-seed](https://github.com/NativeScript/nativescript-plugin-seed) so you do not need to care about anything.
+
+Let's write a few additional tests to default ones as a showcase. We will continue from the point where [Building UI Plugins Using Composite Components]({% building-ui-plugins-composite-components %}) article left us so be sure you are aware of it. In order to test the three properties defined there we will write a test for each of them. Each test will be in a separate suite.
```
Comment (tjvantoll) [inline plugins/unit-tests.md]: Let’s add a few additional tests to the `tests.js` file.

File: plugins/unit-tests.md [hunk 10]
```diff
@@ -0,0 +1,92 @@
...
+});
+
+describe("bottomText property", function() {
+    it("value is applied to bottom label", function() {
+        uiPlugin.bottomText = "gain";
+        expect(uiPlugin.getChildAt(0).getChildAt(2).text).toEqual("gain");
+    });
+});
+```
+Every test assigns a value to the property in testing and verifies that the same value is applied to the element in the visual tree that uses it. I guess you wonder how I know the indexes of the elements selected with `getChildAt()` function. The visual tree of the [nativescript-ui-plugin](https://github.com/NativeScript/nativescript-ui-plugin) in our example is pretty simple. It has a grid layout containing three elements which makes it easy to orientate in the structure. In case of more complicated plugin I would suggest you to use some of the [LayoutBase](https://docs.nativescript.org/api-reference/classes/_ui_layouts_layout_base_.layoutbase.html) class methods to explore the visual three. For example:
```
Comment (tjvantoll) [inline plugins/unit-tests.md]: > I guess you wonder how I know the indexes of the elements selected with `getChildAt()` function.

I would rephrase this to explain what `getChildAt()` does and link to the API documentation.

File: plugins/unit-tests.md [hunk 11]
```diff
@@ -0,0 +1,92 @@
...
+uiElement.eachChildView((view)=>{
+    console.log("======START======");
+    console.log("Index: " + uiElement.getChildIndex(view));
+    console.log("Element: " + view);
+    console.log("======END======");
+    console.log(" ");
+});
+```
+
+The complete `test.js` file you can find [here](https://github.com/NativeScript/nativescript-ui-plugin/blob/master/demo/app/tests/tests.js).
```
Comment (tjvantoll) [inline plugins/unit-tests.md]: You can find the complete `tests.js` file here.

File: plugins/unit-tests.md [hunk 12]
```diff
@@ -0,0 +1,92 @@
...
+We have our tests ready and it is time to get them in action. Fortunately, [nativescript-plugin-seed](https://github.com/NativeScript/nativescript-plugin-seed) has already provided the commands we need in the form of npm scripts. Just navigate to your `src` folder and run:
+
+```
+npm run test.android
+npm run test.ios
+```
+
+Be sure that you have available android/ios device or simulator.
+
+> **NOTE**: Using the npm scripts to run your tests is the best option, but in case of debugging purposes, where you want to output some console logs, you will have to navigate to `demo` folder and run `tns test android` or `tns test ios`.
```
Comment (tjvantoll) [inline plugins/unit-tests.md]: to your `demo` folder

File: plugins/unit-tests.md (hunk 4 above)
Comment (tjvantoll) [inline plugins/unit-tests.md]: you get an out-of-the-box fully-functional `.travis.yml` file

activate your repository, and make sure `Build pushes` and `Build pull requests` are enabled

**Review by tjvantoll — COMMENTED** [review]
Looks good. One note: can you change the “Adding unit tests” link at the bottom of https://docs.nativescript.org/plugins/building-plugins to point at this article?

File: plugins/unit-tests.md (hunk 1 above)
Comment (zbranzov) [reply plugins/unit-tests.md]: @radeva not sure about angular so will leave it this way for now

**Review by radeva — APPROVED** [review]
